### PR TITLE
Update rust version to 1.74

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3621** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3621" target="_blank">Graphite</a> 
    * **PR #3622** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3622" target="_blank">Graphite</a> 
      * **PR #3623** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3623" target="_blank">Graphite</a> 
        * **PR #3619** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3619" target="_blank">Graphite</a> 
          * **PR #3626** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3626" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3626?utm_source=stack-comment).

File: crypto/hashes/Cargo.toml [hunk 1]
```diff
@@ -15,8 +15,4 @@ mc-crypto-digestible = { path = "../digestible", default_features = false }
 
 digest = { version = "0.10.1", default-features = false }
 
-[target.'cfg(any(target_feature = "avx2", target_feature = "avx"))'.dependencies]
-blake2 = { version = "0.10.2", default-features = false, features = ["simd"] }
-
-[target.'cfg(not(any(target_feature = "avx2", target_feature = "avx")))'.dependencies]
-blake2 = { version = "0.10.2", default-features = false }
+blake2 = { version = "0.10.6", default-features = false }
```
Comment (nick-mobilecoin) [inline crypto/hashes/Cargo.toml:18]: TL;DR I'm not convinced the `simd` feature was originally justified in its use.

This change of dropping simd on `avx` architectures is probably the most questionable change in this pr.
I ran the blake2 benchmarks on my intel laptop, which other than the 256 10000 run, the runs were within the error bars of each other.
```
 blake2 git:(master) cargo +nightly-2023-01-22 bench          

...

running 8 tests
test blake2b512_10    ... bench:           9 ns/iter (+/- 0) = 1111 MB/s
test blake2b512_100   ... bench:          89 ns/iter (+/- 0) = 1123 MB/s
test blake2b512_1000  ... bench:         856 ns/iter (+/- 4) = 1168 MB/s
test blake2b512_10000 ... bench:       8,524 ns/iter (+/- 26) = 1173 MB/s
test blake2s256_10    ... bench:          14 ns/iter (+/- 0) = 714 MB/s
test blake2s256_100   ... bench:         142 ns/iter (+/- 1) = 704 MB/s
test blake2s256_1000  ... bench:       1,376 ns/iter (+/- 6) = 726 MB/s
test blake2s256_10000 ... bench:      13,717 ns/iter (+/- 173) = 729 MB/s
```
and
```
blake2 git:(master) cargo +nightly-2023-01-22 bench --features simd

...

running 8 tests
test blake2b512_10    ... bench:          10 ns/iter (+/- 0) = 1000 MB/s
test blake2b512_100   ... bench:          98 ns/iter (+/- 0) = 1020 MB/s
test blake2b512_1000  ... bench:         959 ns/iter (+/- 10) = 1042 MB/s
test blake2b512_10000 ... bench:       9,599 ns/iter (+/- 43) = 1041 MB/s
test blake2s256_10    ... bench:          14 ns/iter (+/- 0) = 714 MB/s
test blake2s256_100   ... bench:         140 ns/iter (+/- 0) = 714 MB/s
test blake2s256_1000  ... bench:       1,388 ns/iter (+/- 5) = 720 MB/s
test blake2s256_10000 ... bench:      13,879 ns/iter (+/- 48) = 720 MB/s
```
I also ran it on a HW vm
```
➜  blake2 git:(master) cargo +nightly-2023-01-22 bench                

...

running 8 tests
test blake2b512_10    ... bench:           9 ns/iter (+/- 1) = 1111 MB/s
test blake2b512_100   ... bench:          94 ns/iter (+/- 5) = 1063 MB/s
test blake2b512_1000  ... bench:         888 ns/iter (+/- 159) = 1126 MB/s
test blake2b512_10000 ... bench:       8,794 ns/iter (+/- 1,812) = 1137 MB/s
test blake2s256_10    ... bench:          16 ns/iter (+/- 2) = 625 MB/s
test blake2s256_100   ... bench:         148 ns/iter (+/- 33) = 675 MB/s
test blake2s256_1000  ... bench:       1,488 ns/iter (+/- 168) = 672 MB/s
test blake2s256_10000 ... bench:      14,494 ns/iter (+/- 2,635) = 689 MB/s
```
with simd
```
➜  blake2 git:(master) cargo +nightly-2023-01-22 bench --features simd

...

running 8 tests
test blake2b512_10    ... bench:           9 ns/iter (+/- 1) = 1111 MB/s
test blake2b512_100   ... bench:          94 ns/iter (+/- 6) = 1063 MB/s
test blake2b512_1000  ... bench:         877 ns/iter (+/- 254) = 1140 MB/s
test blake2b512_10000 ... bench:       8,852 ns/iter (+/- 1,008) = 1129 MB/s
test blake2s256_10    ... bench:          14 ns/iter (+/- 1) = 714 MB/s
test blake2s256_100   ... bench:         131 ns/iter (+/- 16) = 763 MB/s
test blake2s256_1000  ... bench:       1,369 ns/iter (+/- 211) = 730 MB/s
test blake2s256_10000 ... bench:      13,231 ns/iter (+/- 1,984) = 755 MB/s
```
While the average values are somewhat faster for simd, it's only for the blake2s256 10000 run and the error bars overlap. With these inconsistencies between HW and the closeness of the timings, I question if `simd` was added as a visible improvement or if someone added more of "it sounds faster"

File: sgx/panic/src/lib.rs [hunk 2]
```diff
@@ -9,6 +9,7 @@
 //
 // Alloc feature must be enabled to get APIs associated to catching panics
 // and rethrowing them, because in Rust those APIs use the Box type.
+#![allow(internal_features)]
```
Comment (nick-mobilecoin) [inline sgx/panic/src/lib.rs:12]: This is kind of an annoying lint. It barks on the use of `feature(lang_items)` but that's the only way to the `eh_personality` that is recommended for customizing panic behavior...

File: crypto/hashes/Cargo.toml (hunk 1 above)
Comment (eranrund) [reply crypto/hashes/Cargo.toml:18]: Good chance this just seemed like a "good to have" but was never actually benchmarked, and the benchmarks you provide do show the `simd` feature isn't making much of a difference.

**Review by eranrund — APPROVED** [review]
LGTM, would be nice to see how this perform when deployed compared to our current rustc version. Unfortunately since there's no functioning CD I guess this will have to wait...

**Pull request merged**